packages/react-dom/src/events/__tests__/SelectEventPlugin-test.internal.js

```diff
@@ -58,34 +68,30 @@ describe('SelectEventPlugin', () => {
     expect(mouseup).toBe(null);
   });
 
-  it('should extract if an `onSelect` listener is present', () => {
-    class WithSelect extends React.Component {
-      render() {
-        return <input type="text" onSelect={this.props.onSelect} />;
-      }
-    }
-
-    var cb = jest.fn();
-
-    var rendered = ReactTestUtils.renderIntoDocument(
-      <WithSelect onSelect={cb} />,
+  it('should register event only if the `onSelect` listener is present', () => {
+    var select = jest.fn();
+    var onSelect = event => {
+      expect(typeof event).toBe('object');
+      expect(event.type).toBe('select');
+      expect(event.target).toBe(node);
+      select(event.currentTarget);
+    };
+
+    var node = ReactDOM.render(
+      <input type="text" onSelect={onSelect} />,
+      container,
     );
-    var node = ReactDOM.findDOMNode(rendered);
-
-    node.selectionStart = 0;
-    node.selectionEnd = 0;
     node.focus();
+    expect(select.mock.calls.length).toBe(0);
 
-    var focus = extract(node, 'topFocus');
-    expect(focus).toBe(null);
+    var nativeEvent = document.createEvent('Event');
```

Comment:
Let's use the newer API for this? e.g. `new MouseEvent('mousedown', {bubbles: true, cancelable: true})`